testing/kuttl/e2e/exporter-password-change/04-assert.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
+commands:
+# Ensure that the password has been updated in the exporter and it can still access
+# Postgres. 
+# - Check that the DATA_SOURCE_PASS env var is set to `password`. Changing an env
+#   var means that the container has been restarted
+# - Check that all containers in the instance pod are ready after restart
+# - Grab the scrape_error output from exporter metrics and check that there were no scrape errors
+- script: |
+    retry() { bash -ceu 'printf "$1\nSleeping...\n" && sleep 5' - "$@"; }
+    contains() { bash -ceu '[[ "$1" == *"$2"* ]]' - "$@"; }
+    check_containers_ready() { bash -ceu ' \
+      echo "$1" | jq -e ".[] | select(.type==\"ContainersReady\") | .status==\"True\""' - "$@";
+    }
+    pod=$(kubectl get pods -o name -n $NAMESPACE \
+      -l postgres-operator.crunchydata.com/cluster=exporter-password-change \
+      -l postgres-operator.crunchydata.com/crunchy-postgres-exporter=true)
+    [ "$pod" = "" ] && retry "Pod not found" && exit 1
+    password=$(kubectl exec -n ${NAMESPACE} ${pod} -c exporter -- bash -c 'echo $DATA_SOURCE_PASS')
+    { contains "${password}" "password"; } || {
+      retry "unexpected password: ${password}"
+      exit 1
+    }
+    condition_json=$(kubectl get ${pod} -n ${NAMESPACE} -o jsonpath="{.status.conditions}")
+    [ "$condition_json" = "" ] && retry "conditions not found" && exit 1
+    { check_containers_ready "$condition_json"; } || {
+      retry "containers not ready"
+      exit 1
+    }
+    scrape_metrics=$(kubectl exec ${pod} -c exporter -n ${NAMESPACE} -- \
+      curl --silent http://localhost:9187/metrics | grep "pg_exporter_last_scrape_error")
+    { contains "${scrape_metrics}" 'pg_exporter_last_scrape_error 0'; } || {
+      retry "${scrape_metrics}" && exit 1
```

Comment:
```suggestion
      retry "${scrape_metrics}"
      exit 1
```

Reply:
:+1: I'll push this up in the pass file work